src/gui/sharedialog.cpp

```diff
@@ -0,0 +1,308 @@
+#include "sharedialog.h"
+#include "ui_sharedialog.h"
+#include "networkjobs.h"
+#include "account.h"
+#include "json.h"
+#include "folderman.h"
+#include <QBuffer>
+#include <QMovie>
+
+namespace {
+    int SHARETYPE_PUBLIC = 3;
+}
+
+namespace OCC {
+
+ShareDialog::ShareDialog(QWidget *parent) :
+    QDialog(parent),
+    _ui(new Ui::ShareDialog)
+{
+    setAttribute(Qt::WA_DeleteOnClose);
+    _ui->setupUi(this);
+    _ui->pushButton_copy->setIcon(QIcon::fromTheme("edit-copy"));
+    layout()->setSizeConstraint(QLayout::SetFixedSize);
+    QMovie *movie = new QMovie("/home/azelphur/ownCloud-share-tools/loading-icon.gif");
+    movie->start();
+    _ui->labelShareSpinner->setMovie(movie);
+    _ui->labelShareSpinner->hide();
+
+    _ui->labelPasswordSpinner->setMovie(movie);
+    _ui->labelPasswordSpinner->hide();
+
+    _ui->labelCalendarSpinner->setMovie(movie);
+    _ui->labelCalendarSpinner->hide();
+    connect(_ui->checkBox_shareLink, SIGNAL(clicked()), this, SLOT(slotCheckBoxShareLinkClicked()));
+    connect(_ui->checkBox_password, SIGNAL(clicked()), this, SLOT(slotCheckBoxPasswordClicked()));
+    connect(_ui->lineEdit_password, SIGNAL(returnPressed()), this, SLOT(slotPasswordReturnPressed()));
+    connect(_ui->checkBox_expire, SIGNAL(clicked()), this, SLOT(slotCheckBoxExpireClicked()));
+    connect(_ui->calendar, SIGNAL(clicked(QDate)), SLOT(slotCalendarClicked(QDate)));
+
+    _ui->labelShareSpinner->hide();
+    _ui->lineEdit_shareLink->hide();
+    _ui->pushButton_copy->hide();
+    _ui->lineEdit_password->hide();
+    _ui->checkBox_password->hide();
+    _ui->checkBox_expire->hide();
+    _ui->calendar->hide();
+    _ui->lineEdit_shareGroup->setPlaceholderText("Share with group...");
+    _ui->lineEdit_shareUser->setPlaceholderText("Share with user...");
+    _ui->lineEdit_password->setPlaceholderText("Choose a password for the public link");
+}
```

Comment:
All user-readable text must be wrapped in `tr()`. Otherwise, it's not translatable.